### PR TITLE
Add support for integer number to varchar coercer in unpartitioned table

### PR DESCRIPTION
**Pull request by guiyanakuang**
## Description
Add support for integer number to varchar coercer in unpartitioned table.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Section
* Add support for integer number to varchar coercer in unpartitioned table
```

### CODE REVIEW

**Comment by guiyanakuang** [comment]
Hi @Praveen2112, could you please review this PR when you get a chance? Thanks!

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java [hunk 1]
```diff
@@ -98,10 +98,21 @@ protected void doTestHiveCoercion(HiveTableDefinition tableDefinition)
                 "tinyint_to_smallint",
                 "tinyint_to_int",
                 "tinyint_to_bigint",
+                "tinyint_to_string",
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:101]: Is it possible to split the commits into two where we add additional test cases to `BaseTestHiveCoercion` and ensure that they fail for unpartitioned tables and then apply the fix for orc type translator ?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java [hunk 2]
```diff
@@ -14,15 +14,24 @@
 package io.trino.plugin.hive.orc;
 
 import io.trino.orc.metadata.OrcType.OrcTypeKind;
+import io.trino.plugin.hive.coercions.IntegerNumberToVarcharCoercer;
```
Comment (Praveen2112) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java:17]: Can we reowrd the commit message as `Add support for integer number to varchar coercer in unpartitioned table` and in the commit description we could mention them in detail

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java [hunk 3]
```diff
@@ -157,10 +168,21 @@ protected void insertTableRows(String tableName, String floatToDoubleType)
                         "  TINYINT '-1', " +
                         "  TINYINT '2', " +
                         "  TINYINT '-3', " +
+                        "  TINYINT '4', " +
+                        "  TINYINT '-5', " +
+                        "  TINYINT '6', " +
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:173]: Can we provide a two digit value for tinyint ?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java [hunk 4]
```diff
@@ -816,10 +904,21 @@ private static void alterTableColumnTypes(String tableName)
...
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_string tinyint_to_string string", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_bounded_varchar tinyint_to_bounded_varchar varchar(30)", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_smaller_varchar tinyint_to_smaller_varchar varchar(3)", tableName));
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_int smallint_to_int int", tableName));
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_bigint smallint_to_bigint bigint", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_string smallint_to_string string", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_bounded_varchar smallint_to_bounded_varchar varchar(30)", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_smaller_varchar smallint_to_smaller_varchar varchar(5)", tableName));
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:914]: When we are coercing to smaller column we need to ensure that the value would be truncated so for a tinyint of `59` - coercing to varchar(1) should give `5` as its output.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java (hunk 2 above)
Comment (guiyanakuang) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java:17]: Done

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java (hunk 3 above)
Comment (guiyanakuang) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java]: Done

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java (hunk 4 above)
Comment (guiyanakuang) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:914]: Done

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java (hunk 1 above)
Comment (guiyanakuang) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:101]: Done

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java (hunk 4 above)
Comment (guiyanakuang) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java:914]: Thanks @Praveen2112 for the review, I modified `IntegerNumberToVarcharCoercer` to support truncation.

**Comment by guiyanakuang** [comment]
CI hit https://github.com/trinodb/trino/issues/18392

**Comment by mosabua** [comment]
:wave: @guiyanakuang and @Praveen2112  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by Praveen2112** [comment]
Lets' restrict truncation for coercion today. Sorry for the initial confusion

**Pull request closed without merging**